### PR TITLE
Search filter improvements

### PR DESCRIPTION
**Pull request by openjck**
- [x] filters are not localized
- [x] RTL display bugs
- [x]  bugs with long queries
- [x] top of filter blue boxes are being cut off by container
- [x] L/R scrolling for long filter combinations not being triggered
- [x] default size of search input is too small for German placeholder text

### CODE REVIEW

**Comment by openjck** [comment]
If anyone feels the urge, you can make further improvements by committing to [mozilla/bug-1100690-search-filter-improvements](https://github.com/mozilla/kuma/tree/bug-1100690-search-filter-improvements). No pressure, though.

File: kuma/landing/templates/landing/homepage.html [hunk 1]
```diff
@@ -44,13 +44,35 @@
...
-  </form>
+    {% if waffle.flag('search_suggestions') %}
+        <form id="home-search-form" action="{{ url('search') }}" method="get" class="search nodisable">
+            <div class="home-search-form search-form">
+                <div class="search-filter-wrapper">
+                    <div id="filters-right" class="filter-arrow">
+                        <i class="icon-chevron-left" aria-hidden="true"></i>
+                    </div>
+                    <div class="filters"></div>
+                    <div id="filters-left" class="filter-arrow">
```
Comment (stephaniehobson) [inline kuma/landing/templates/landing/homepage.html]: This and filters-right should probably be buttons not <divs>, OR we can update the js to add tabindex, aria roles, and an event listener... I think buttons is easier ;)

File: kuma/landing/templates/landing/homepage.html [hunk 2]
```diff
@@ -44,13 +44,35 @@
...
+                        <i class="icon-chevron-right" aria-hidden="true"></i>
```
Comment (stephaniehobson) [inline kuma/landing/templates/landing/homepage.html]: Needs fall back text for the icon.

File: kuma/landing/templates/landing/homepage.html [hunk 3]
```diff
@@ -44,13 +44,35 @@
...
+                    </div>
+                    <label for="home-q" class="offscreen">{{ _('Search') }}</label>
+                    <input class="search-input" type="search" id="home-q" name="q" placeholder="{{ _('Search the docs') }}" />
+                    <button type="submit" class="offscreen">{{ _('Search') }}</button>
+                    <button class="show-topics" type="button">
+                        <i class="icon-caret-down" aria-hidden="true"></i>
```
Comment (stephaniehobson) [inline kuma/landing/templates/landing/homepage.html]: Icon needs fallback text.

File: media/stylus/base/elements/forms.styl [hunk 4]
```diff
@@ -78,7 +78,7 @@ label {
 
     &.neutral {
         background-color: #0095dd;
-        box-shadow: inset 0 -1px #00539f;
+        box-shadow: inset 0 -1px $homepage-background-color;
```
Comment (stephaniehobson) [inline media/stylus/base/elements/forms.styl]: Yay variables!

Comment (openjck) [inline media/stylus/base/elements/forms.styl]: :tada:

File: media/js/search-suggestions.js [hunk 5]
```diff
@@ -112,7 +80,7 @@
               },
               showarrows: function() {
                 // max-width of $searchFilters is 70% in search-suggestions.styl
-                var max_width = 69 / 100 * $search.width();
+                var max_width = 69 / 100 * $searchForm.width();
```
Comment (stephaniehobson) [inline media/js/search-suggestions.js]: Something is wrong with the math here. I can't trigger it by adding filters. Maybe we want the width of .search-filter-wrapper instead?

File: media/stylus/search-suggestions.styl [hunk 6]
```diff
@@ -1,181 +1,152 @@
...
-        display: block;
+    .show-topics {
+         background-color: rgba-fallback(rgba(255, 255, 255, 0)) !important;
+         box-shadow: none;
+         color: $text-color !important;
+         font-size: 1.2em !important;
+         position: absolute;
+         right: 0;
+         top: 3px;
+     }
```
Comment (stephaniehobson) [inline media/stylus/search-suggestions.styl]: Spacing in this declaration is a bit off.

**Comment by stephaniehobson** [comment]
If you type a really long query you get some odd behaviour. Should be able to fix by not letting it keep getting longer after it hits the width of the search wrapper

**Comment by stephaniehobson** [comment]
The javascript file has a lot of places where there are 2 space indents instead of 4 :/

**Comment by stephaniehobson** [comment]
Mostly nitpicks, hopefully nothing too big.

File: media/stylus/search-suggestions.styl (hunk 6 above)
Comment (openjck) [inline media/stylus/search-suggestions.styl]: Oh huh, you're right. Good eye. :smile:

**Comment by openjck** [comment]
> The javascript file has a lot of places where there are 2 space indents instead of 4 :/

Good catch. I can probably get Vim to take care of that without much effort.

**Comment by stephaniehobson** [comment]
Bah, I forgot to review in RTL, sorry. Will do that in a bit.

**Comment by stephaniehobson** [comment]
This goes kind of weirdly awry in RTL.
screen shot 2015-04-13 at 11 58 32

Also, I noticed that the top 2px of the filters are being cut off in any language. You can tell by looking at the rounded corners, the bottom corners are rounded but the top ones are not, but it's because they're being clipped by the container.

**Comment by stephaniehobson** [comment]
Working on rebase and RTL fixes.

**Comment by openjck** [comment]
\o/

**Comment by stephaniehobson** [comment]
screen shot 2015-04-21 at 15 06 53

**Comment by openjck** [comment]
So awesome! :v:

**Comment by openjck** [comment]
@stephaniehobson: Does that address all of your earlier line comments?

**Comment by stephaniehobson** [comment]
@openjck Yup!

What we need now is someone to look at `search_filters` in `context_processors.py` to figure out why it's not returning localized results. That's the last blocker I see. Not sure we're going to be able to wrangle any of the back-end team into doing that though.

**Comment by openjck** [comment]
Maybe @robhudson would know?

**Comment by robhudson** [comment]
Updated to add l10n for the search filters.

**Comment by stephaniehobson** [comment]
:+1: This is ready to go!

**Comment by stephaniehobson** [comment]
Apparently it needs @openjck to review the RTL changes first...

**Pull request merged**